thumberd/thumberd.go

```diff
@@ -497,9 +498,18 @@ func thumbServer(w http.ResponseWriter, r *http.Request, sem chan int) {
 	w.Header().Set("Content-Type", content_type) //
 	w.Header().Set("Last-Modified", time.Now().UTC().Format(http.TimeFormat))
 
+	imageBlob, err := fetchImageWithCorrectFormat(srcReader.Body)
+	if err != nil {
```

Comment:
Is my understanding correct that we can reduce wait time by filtering out invalid images earlier? Are there lots of such invalid images?

Reply:
No much but it hurts the performance badly. The early detect logic is in `fetchImageWithCorrectFormat`. I read first 20 bytes and detect the format, then read the remaining bytes.